app/frontend/src/features/userQueries/useUsers.ts

```diff
@@ -0,0 +1,54 @@
+import { Error } from "grpc-web";
+import { useEffect } from "react";
+import { useQueries, useQueryClient } from "react-query";
+import { User } from "../../pb/api_pb";
+import { service } from "../../service";
+import { userStaleTime } from "./constants";
+
+export default function useUsers(ids: number[], invalidate: boolean = false) {
+  const queryClient = useQueryClient();
+  useEffect(() => {
+    if (invalidate) {
+      queryClient.invalidateQueries({
+        predicate: (query) =>
+          query.queryKey[0] === "user" &&
+          ids.indexOf(query.queryKey[1] as number) > -1,
```

Comment:
```suggestion
          ids.find(query.queryKey[1] as number),
```
I think using `.find` and check for truthiness if the id exists is a bit clearer to read.